src/main/webapp/app/shared/participant-scores/participant-scores.service.ts

```diff
@@ -0,0 +1,120 @@
+import { HttpClient, HttpParams, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { SERVER_API_URL } from 'app/app.constants';
+import { Observable } from 'rxjs';
+
+/**
+ * Normally the use of null in the client is discouraged but in this case it is fine as the server will always send all
+ * the properties but will give them the value null (a property will never be undefined here).
+ */
+export class ParticipantScoreDTO {
+    public id: number | null;
+    public userId: number | null;
+    public userName: string | null;
+    public teamId: number | null;
+    public teamName: string | null;
+    public exerciseId: number | null;
+    public exerciseTitle: string | null;
+    public lastResultId: number | null;
+    public lastResultScore: number | null;
+    public lastRatedResultId: number | null;
+    public lastRatedResultScore: number | null;
+}
+
+/**
+ * Normally the use of null in the client is discouraged but in this case it is fine as the server will always send all
+ * the properties but will give them the value null (a property will never be undefined here).
+ */
+export class ParticipantScoreAverageDTO {
+    public userName: string | null;
+    public teamName: string | null;
+    public averageScore: number | null;
+    public averageRatedScore: number | null;
```

Comment:
also here?

Reply:
Yes. In this case here yes. I wrote the server code such that all the properties are sent always. The object is used in the table as is, just dropping some properties makes it harder to just display everything in a table. (i would have to always handle the case that every single property could potentially be undefined and so on)